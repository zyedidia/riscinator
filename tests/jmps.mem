03f00d13
043d7113
04016193
0401c213
fff06293
00002313
00a02393
00400413
01a42023
fe242e23
00342223
00042483
ffc42503
00442583
0180006f
00000013
00100613
0180006f
00000013
00100613
ff5ff06f
00000013
00100613
00039693
07068693
00068067
00000013
00100713
00100793
00c000ef
00000013
00100813
00100893
00000663
00000013
00100913
00100993
000d0663
00000013
00100a13
00100a93
00001663
00000013
00100b13
00100b93
000d1663
00000013
00100c13
00100c93
00000013
