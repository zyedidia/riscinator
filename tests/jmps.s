.section ".text.boot"

.globl _start
_start:
		addi x26,   x0,    63  # x26 = 63
		andi x2,   x26,    67  # x2 = 3
		ori  x3,   x2,    64  # x3 = 67
		xori x4,   x3,    64  # x4 = 3
		ori  x5,   x0,    -1  # x5 = -1
		slti x6,   x0,    0   # x6 = 0
		slti x7,   x0,    10  # x7 = 1
		slli x8,   x7,    20  # x8 = 0x100000
		addi x8,   x8,    4   # x8 = 0x100004
		sw   x26,   0(x8)      # 0x4 = 63
		sw   x2,   -4(x8)     # 0x0 = 3
		sw   x3,   4(x8)      # 0x8 = 67
		lw   x9,   0(x8)      # x9 = 63
		lw   x10,  -4(x8)     # x10 = 3
		lw   x11,  4(x8)      # x11 = 67
		j    A
		addi x12,  x0,    1
B:      j    C
		addi x12, x0,    1
A:      j    B
		addi x12,  x0,    1
C:      slli  x13,  x7,    20
		addi x13,  x13,   112
		jr   x13 # j D
		addi x14,  x0,    1
		addi x15,  x0,    1
		jal  D
		addi x16,  x0,    1
D:      addi x17,  x0,    1
		beq  x0,   x0,    E
		addi x18,  x0,    1
E:      addi x19,  x0,    1
		beq  x26,   x0,    F
		addi x20,  x0,    1
F:      addi x21,  x0,    1
		bne  x0,   x0,    G
		addi x22,  x0,    1
G:      addi x23,  x0,    1
		bne  x26,   x0,    H
		addi x24,  x0,    1
H:      addi x25,  x0,    1
done:   j done
