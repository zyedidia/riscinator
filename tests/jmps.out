x0: 0
x1: 120
x2: 3
x3: 67
x4: 3
x5: -1
x6: 0
x7: 1
x8: 4
x9: 63
x10: 3
x11: 67
x12: 1
x13: 113
x14: 0
x15: 1
x16: 0
x17: 0
x18: 0
x19: 1
x20: 1
x21: 1
x22: 1
x23: 1
x24: 0
x25: 1
x26: 63
x27: 0
x28: 0
x29: 0
x30: 0
x31: 0
[0x0]: 3
[0x4]: 63
[0x8]: 67

